spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/procedures/AddFilesProcedure.java

```diff
@@ -170,7 +173,8 @@ private void importFileTable(
 
     if (table.spec().isUnpartitioned()) {
       Preconditions.checkArgument(
-          partitions.isEmpty(), "Cannot add partitioned files to an unpartitioned table");
+          !Spark3Util.isPartitioned(spark(), tableLocation),
```

Comment:
`partitions.isEmpty()` will return true when there are partition predicates which filter out of partition. Hence we have have changed this condition?

Reply:
As we now rely on table's partitionSpec to get the partition paths, if a table is unpartitioned `partitions.isEmpty` will always be true. Hence we need this check for unpartitioned tables to do a quick check if the path contains any dir.